client/rest-high-level/src/main/java/org/elasticsearch/client/rollup/GetRollupJobResponse.java

```diff
@@ -0,0 +1,418 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client.rollup;
+
+import org.elasticsearch.client.rollup.job.config.RollupJobConfig;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.ObjectParser;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Objects;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.constructorArg;
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.optionalConstructorArg;
+import static java.util.Collections.unmodifiableList;
+
+/**
+ * Response from rollup's get jobs api.
+ */
+public class GetRollupJobResponse implements ToXContentObject {
+    private static final ParseField JOBS = new ParseField("jobs");
+    private static final ParseField CONFIG = new ParseField("config");
+    private static final ParseField STATS = new ParseField("stats");
+    private static final ParseField STATUS = new ParseField("status");
+    private static ParseField NUM_PAGES = new ParseField("pages_processed");
+    private static ParseField NUM_INPUT_DOCUMENTS = new ParseField("documents_processed");
+    private static ParseField NUM_OUTPUT_DOCUMENTS = new ParseField("rollups_indexed");
+    private static ParseField NUM_INVOCATIONS = new ParseField("trigger_count");
+    private static final ParseField STATE = new ParseField("job_state");
+    private static final ParseField CURRENT_POSITION = new ParseField("current_position");
+    private static final ParseField UPGRADED_DOC_ID = new ParseField("upgraded_doc_id");
+
+    private List<JobWrapper> jobs;
+
+    GetRollupJobResponse(final List<JobWrapper> jobs) {
+        this.jobs = Objects.requireNonNull(jobs, "jobs is required");
+    }
+
+    /**
+     * Jobs returned by the request.
+     */
+    public List<JobWrapper> getJobs() {
+        return jobs;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        final GetRollupJobResponse that = (GetRollupJobResponse) o;
+        return jobs.equals(that.jobs);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(jobs);
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
```

Comment:
hey, these toXContent are not needed in responses. They help with the tests but you can manually create a parser for now, as I have to think about how to better do these tests without putting the toXContent logic in the classes themselves.

Reply:
KK. I'll pull.

Reply:
Doesn't this make testing a lot harder than just implementing a manual parser?  E.g. we can't use `AbstractXContentTestCase` anymore, which does all the xcontent shuffling serializing/reparsing/equivalence checks?